tests/functional/test_s3.py

```diff
@@ -2141,6 +2141,29 @@ def test_checksums_included_in_expected_operations(
         assert "Content-MD5" in stub.requests[-1].headers
 
 
+@pytest.mark.parametrize(
+    "content_encoding, expected_header",
+    [("foo", "foo,aws-chunked"), (None, "aws-chunked")],
+)
+def test_checksum_content_encoding(content_encoding, expected_header):
+    op_kwargs = {
+        "Bucket": "mybucket",
+        "Key": "mykey",
+        "Body": b"foo",
+        "ChecksumAlgorithm": "sha256",
+    }
+    if content_encoding is not None:
+        op_kwargs["ContentEncoding"] = content_encoding
+    s3 = _create_s3_client()
+    with ClientHTTPStubber(s3) as http_stubber:
+        http_stubber.add_response()
+        s3.put_object(**op_kwargs)
+        assert (
+            http_stubber.requests[-1].headers["Content-Encoding"].decode()
+            == expected_header
+        )
```

Comment:
We probably still want to clean this up. It's going to be difficult to understand the failure because we've put multiple failure cases onto a single line.

```suggestion
        request_headers = http_stubber.requests[-1].headers
        assert request_headers["Content-Encoding"] == expected_header
```

Reply:
And just in case it's missed, we need the expected_header to be bytes.

Reply:
Ok I'll shoot that up now.